code/modules/crafting/craft.dm

```diff
@@ -1,24 +1,35 @@
 /datum/personal_crafting
 	var/busy
 	var/viewing_category = 1 //typical powergamer starting on the Weapons tab
-	var/list/categories = list(CAT_WEAPON,
-				CAT_AMMO,
+	var/viewing_subcategory = 1
+	var/list/categories = list(
+				CAT_WEAPONRY,
 				CAT_ROBOT,
 				CAT_MISC,
 				CAT_PRIMAL,
-				CAT_BREAD,
-				CAT_BURGER,
-				CAT_CAKE,
-				CAT_EGG,
-				CAT_MEAT,
-				CAT_MISCFOOD,
-				CAT_PASTRY,
-				CAT_PIE,
-				CAT_PIZZA,
-				CAT_SALAD,
-				CAT_SANDWICH,
-				CAT_SOUP,
-				CAT_SPAGHETTI)
+				CAT_FOOD)
+	var/list/subcategories = list(
```

Comment:
I feel like this is a weird way to do this; is there a reason categories can't just be an associated list, like so:

    	var/list/categories = list( //Category = list(subcategories) or CAT_NONE if it has no subcategories
				CAT_WEAPONRY = list(CAT_WEAPON, CAT_AMMO),
 				CAT_ROBOT = CAT_NONE,
 				CAT_MISC,
 				CAT_MISC = CAT_NONE,
 				CAT_PRIMAL = CAT_NONE,
				CAT_FOOD = list(CAT_BREAD, CAT_BURGER, CAT_CAKE, CAT_EGG, CAT_MEAT, CAT_MISCFOOD,
							CAT_PASTRY, CAT_PIE, CAT_PIZZA, CAT_SALAD, CAT_SANDWICH, CAT_SOUP, 
							CAT_SPAGHETTI))